src/main/java/ManagerClasses/InputManager.java

```diff
@@ -0,0 +1,78 @@
+package ManagerClasses;
+
+import CommandClasses.Command;
+import ExceptionClasses.EmptyByOrAtWhenException;
+import ExceptionClasses.EmptyTaskDescriptionException;
+
+import java.util.Scanner;
+
+public class InputManager {
+    private final TaskManager taskManager;
+
+    public InputManager(){
+        taskManager = new TaskManager();
+    }
+
+    public void manageInput() {
+        taskManager.importTasksFromTxt();
+        while(true) {
+            String input = getUserInput();
+            Command command = new Command(input);
+            switch (command.getCommandType()) {
+            case BYE:
+                PrintManager.printByeMessage();
+                return;
+            case LIST:
+                taskManager.printTasks();
+                break;
+            case DONE:
+                taskManager.markDone(command.getTaskNum());
+                taskManager.saveTasksInTxt();
+                break;
+            case DEADLINE:
+                try {
+                    taskManager.addDeadlineToList(command.getDescription(), command.getWhen());
+                    taskManager.saveTasksInTxt();
+                } catch (EmptyTaskDescriptionException e) {
+                    System.out.println("Error -> Empty task description");
+                } catch (EmptyByOrAtWhenException | NullPointerException e) {
+                    System.out.println("Error -> Deadline by when is not stated");
```

Comment:
This block of code is somewhat lengthy as compared to how the previous cases were handled. Consider if things could be [simplified](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#practice-kissing) further or [SLAP](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) could be incorporated here.